driver-core/src/main/com/mongodb/client/gridfs/model/GridFSFile.java

```diff
@@ -0,0 +1,258 @@
+/*
+ * Copyright 2015 MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.client.gridfs.model;
+
+import org.bson.BsonValue;
+import org.bson.Document;
+import org.bson.types.ObjectId;
+
+import java.util.Date;
+import java.util.List;
+
+import static com.mongodb.assertions.Assertions.notNull;
+
+/**
+ * The GridFSFile
+ *
+ * @since 3.1
+ */
+public final class GridFSFile {
+    private final BsonValue id;
+    private final String filename;
+    private final long length;
+    private final int chunkSize;
+    private final Date uploadDate;
+    private final String md5;
+
+    // Optional values
+    private final Document metadata;
+
+    // Deprecated values
+    private final String contentType;
+    private final List<String> aliases;
```

Comment:
Existing files collection documents created by older implementations of GridFS might have additional unexpected fields at the root level.

What would happen in that case?